botocore/utils.py

```diff
@@ -1085,3 +1085,54 @@ def _get_response(self, full_url, headers, timeout):
 
     def full_url(self, relative_uri):
         return 'http://%s%s' % (self.IP_ADDRESS, relative_uri)
+
+
+def get_configured_signature_version(service_name, client_config,
+                                     scoped_config):
+    """
+    Gets the manually configured signature version.
+
+    :returns: the customer configured signature version, or None if no
+        signature version was configured.
+    """
+    # Client config overrides everything.
+    if client_config and client_config.signature_version is not None:
+        return client_config.signature_version
+
+    # Scoped config overrides picking from the endpoint metadata.
+    if scoped_config is not None:
+        # A given service may have service specific configuration in the
+        # config file, so we need to check there as well.
+        service_config = scoped_config.get(service_name)
+        if service_config is not None and isinstance(service_config, dict):
+            version = service_config.get('signature_version')
+            if version:
+                logger.debug(
+                    "Switching signature version for service %s "
+                    "to version %s based on config file override.",
+                    service_name, version)
+                return version
+    return None
+
+
+def default_s3_presign_to_sigv2(signature_version, signing_name,
```

Comment:
Would it be different? We currently do it for s3 addressing style: https://github.com/boto/botocore/blob/develop/botocore/client.py#L139-L142. For me, I suggested this because:

1) It is weird to have a public function in `utils` that is only used in one location. Typically if it is in `utils`, it should be used by more than one module.

2) By making it internal and not in the `utils` module, it gives us more flexibility to change the interface. I feel that public functions in `utils` is a gray area to break the public interface.